packages/vscode-extension/src/controls/webviewPanel.ts

```diff
@@ -161,6 +120,61 @@ export class WebviewPanel {
     this.panel.webview.html = this.getHtmlForWebview(panelType);
   }
 
+  private async downloadSampleApp(msg: any){
+    ExtTelemetry.sendTelemetryEvent(TelemetryEvent.DownloadSampleStart, { [TelemetryProperty.TriggerFrom]: TelemetryTiggerFrom.Webview, [TelemetryProperty.SampleAppName]: msg.data.appFolder });
+    const folder = await vscode.window.showOpenDialog({
+      canSelectFiles: false,
+      canSelectFolders: true,
+      canSelectMany: false,
+      title: "Select folder to download the sample app",
+    });
+
+    let downloadSuccess = false;
+    if (folder !== undefined) {
+      const sampleAppPath = path.join(folder[0].fsPath, msg.data.appFolder);
+      if (
+        (await fs.pathExists(sampleAppPath)) &&
+        (await fs.readdir(sampleAppPath)).length > 0
+      ) {
+        vscode.window.showErrorMessage(
+          `Path ${sampleAppPath} alreay exists. Select a different folder.`
+        );
+      } else {
+        const dialogManager = DialogManager.getInstance();
+        const progress = dialogManager.createProgressBar("Fetch sample app", 2);
+        progress.start();
+        try {
+          progress.next(`Downloading from '${msg.data.appUrl}'`);
+          const result = await this.fetchCodeZip(msg.data.appUrl);
+          progress.next("Unzipping the sample package");
+          if (result !== undefined) {
+            await this.saveFilesRecursively(
+              new AdmZip(result.data),
+              msg.data.appFolder,
+              folder[0].fsPath
+            );
+            downloadSuccess = true;
+            vscode.commands.executeCommand(
+              "vscode.openFolder",
+              vscode.Uri.file(sampleAppPath)
+            );
+            ext.context.globalState.update("openSampleReadme", true);
+          } else {
+            vscode.window.showErrorMessage("Failed to download sample app");
+          }
+        } finally {
+          progress.end();
+        }
+      }
+    }
+
+    if (downloadSuccess) {
+      ExtTelemetry.sendTelemetryEvent(TelemetryEvent.DownloadSample, { [TelemetryProperty.TriggerFrom]: TelemetryTiggerFrom.Webview, [TelemetryProperty.SampleAppName]: msg.data.appFolder, [TelemetryProperty.Success]: TelemetrySuccess.Yes });
+    } else {
+      ExtTelemetry.sendTelemetryEvent(TelemetryEvent.DownloadSample, { [TelemetryProperty.TriggerFrom]: TelemetryTiggerFrom.Webview, [TelemetryProperty.SampleAppName]: msg.data.appFolder, [TelemetryProperty.Success]: TelemetrySuccess.No });
```

Comment:
I think here you need to sendTelemetryErrorEvent. and compose an error as parameter to it.